dmaws/rds.py

```diff
@@ -295,20 +287,27 @@ def clean_database(self):
                     WHERE (
                         SELECT COUNT(*) FROM supplier_frameworks WHERE supplier_id=suppliers.supplier_id
                     ) = 0
+                    AND supplier_id NOT IN (
+                        SELECT DISTINCT supplier_id FROM services
+                    )
                 ), d1 AS (
                     DELETE FROM contact_information WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 ), d2 AS (
                     DELETE FROM users WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 )
-             DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
+            DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
         """)
 
+        # Remove audit events because we don't use them
         self.log("Delete audit events")
         self.cursor.execute("DELETE FROM audit_events")
 
+        # Remove framework agreements because they contain personal data and we don't rely on them
         self.log("Delete framework agreements")
         self.cursor.execute("DELETE FROM framework_agreements")
 
+        # Overwrite declarations with the smallest possible valid entry
```

Comment:
We use declaration fields for a lot of the close-of-framework scripts, and also in the admin front-end ("nameOfOrganisation", company number and address details are rendered in the countersigning flow.